protocol/pubsub/v2/attributes.go

```diff
@@ -0,0 +1,26 @@
+/*
+ Copyright 2021 The CloudEvents Authors
+ SPDX-License-Identifier: Apache-2.0
+*/
+
+/*
+Package pubsub implements a Pub/Sub binding using google.cloud.com/go/pubsub module
+*/
+package pubsub
+
+import (
+	"context"
+
+	"github.com/cloudevents/sdk-go/v2/binding"
+)
+
+type withCustomAttributes struct{}
+
+func AttributesFrom(ctx context.Context) map[string]string {
+	return binding.GetOrDefaultFromCtx(ctx, withCustomAttributes{}, make(map[string]string)).(map[string]string)
+}
+
+// WithCustomAttributes sets Message Attributes without any Cloud Event logic.
+func WithCustomAttributes(ctx context.Context, attrs map[string]string) context.Context {
```

Comment:
question: do we want this feature per event or as a global behavior on the PubSub protocol implementation (client)?

Reply:
Per event